src/voxel.cpp

```diff
@@ -283,21 +288,25 @@ void Voxel::traverseTree
 bool Voxel::isAtom(const Atom& atom, const Vector& pos_vxl, const double rad_vxl, const double rad_probe){
   Vector dist = pos_vxl - atom.getPosVec();
 
-  if((dist < atom.getRad() - rad_vxl) && (0 < atom.getRad() - rad_vxl)){
+  // TODO the second condition would not be necessary if dist was a double type

```

Comment:
I don't think this is true. I'm not sure, but I think you believe that "dist" is treated as a vector here and that the directionality of the vector is relevant to the comparison. However, my Vector class uses the absolute length every time a comparison is made, such as here. That means that the directionality is irrelevant. The second condition is needed in cases where the voxels are much larger than the atoms, because here the difference between them becomes negative. This has nothing to do with the dist-Vector.

The reason I'm using my Vector class here is that comparisons using the Vector class avoid taking the square root of the length, and use the norm instead, which is more efficient.

Reply:
In cases where the voxel is much larger than the atom, the alternative version proposed in the next line `if(dist + rad_vxl < atom.getRad()){` would be a sufficient condition. Indeed if rad_vxl >> rad_atom, then dist (positive value) + rad_vxl > rad_atom.

Reply:
Oh ok I think now I understand. I was confused because your suggestion `dist + rad_vxl < atom.getRad()` is just a rearrangement of `dist < atom.getRad() - rad_vxl`. The reason why my build requires the second condition, is because in other to avoid the square root, all values are squared. This means negative lengths and positive lengths are no longer distinguished. I think this is a worthy trade off for avoiding the std::pow function, but to determine what is faster you'd have to do testing. I don't think this is worth the time though.